pkg/apis/pipeline/v1/pipeline_validation.go

```diff
@@ -766,23 +757,88 @@ func validateGraph(tasks []PipelineTask) (errs *apis.FieldError) {
 
 func validateMatrix(ctx context.Context, tasks []PipelineTask) (errs *apis.FieldError) {
 	for idx, task := range tasks {
-		errs = errs.Also(task.validateMatrix(ctx).ViaIndex(idx))
+		errs = errs.Also(task.validateMatrix(ctx, tasks).ViaIndex(idx))
```

Comment:
Oh sorry for the confusion, I think we don't need to pass tasks to task.validateMatrix. The validation will be executed num(tasks) times. I was thinking about something like this:

```
func validateMatrix(ctx context.Context, tasks []PipelineTask) (errs *apis.FieldError) {
	for idx, task := range tasks {
		errs = errs.Also(task.validateMatrix(ctx).ViaIndex(idx))
	}
        errs = validateTaskResultsFromMatrixedPipelineTasksConsumed(tasks)
	return errs
}
```